src/IssueViz/IssueVisualizationControl/ViewModels/IssueVisualizationViewModel.cs

```diff
@@ -0,0 +1,207 @@
+
+        public string RuleKey => currentIssue?.Issue?.RuleKey;
+
+        public string RuleHelpLink => string.IsNullOrEmpty(RuleKey) ? null : ruleHelpLinkProvider.GetHelpLink(RuleKey);
+
+        public AnalysisIssueSeverity Severity => currentIssue?.Issue?.Severity ?? AnalysisIssueSeverity.Info;
+
+        public IAnalysisIssueVisualization CurrentIssue
+        {
+            get => currentIssue;
+            private set
+            {
+                currentIssue = value;
+                OnPropertyChanged(null);
+            }
+        }
+
+        public IAnalysisIssueFlowVisualization CurrentFlow
+        {
+            get => currentFlow;
+            set
+            {
+                currentFlow = value;
+
+                if (isBindingUpdatedOutsideOfControl)
+                {
+                    OnPropertyChanged(nameof(CurrentFlow));
+
+                    LocationListItems = BuildLocationListItems(currentFlow);
+                    OnPropertyChanged(nameof(LocationListItems));
+                }
+                else
+                {
+                    selectionEvents.SelectedFlow = currentFlow;
+                }
+            }
+        }
+
+        public IReadOnlyList<ILocationListItem> LocationListItems { get; private set; }
+
+        public LocationListItem CurrentLocationListItem
+        {
+            get => currentLocation;
+            set
+            {
+                currentLocation = value;
+
+                if (isBindingUpdatedOutsideOfControl)
+                {
+                    OnPropertyChanged(nameof(CurrentLocationListItem));
+                }
+                else
+                {
+                    selectionEvents.SelectedLocation = currentLocation?.Location;
+                }
+            }
+        }
+
+        private void SelectionEvents_SelectedIssueChanged(object sender, IssueChangedEventArgs e)
+        {
+            CurrentIssue = e.Issue;
+        }
+
+        private void SelectionEventsOnSelectedFlowChanged(object sender, FlowChangedEventArgs e)
+        {
+            isBindingUpdatedOutsideOfControl = true;
+
+            CurrentFlow = e.Flow;
+
+            isBindingUpdatedOutsideOfControl = false;
+        }
+
+        private void SelectionEvents_SelectedLocationChanged(object sender, LocationChangedEventArgs e)
+        {
+            isBindingUpdatedOutsideOfControl = true;
+
+            if (e.Location == null)
+            {
+                CurrentLocationListItem = null;
+            }
+            else
+            {
+                var selectedLocationListItem = LocationListItems?
+                    .OfType<LocationListItem>()
+                    .FirstOrDefault(x => x.Location == e.Location);
+
+                CurrentLocationListItem = selectedLocationListItem;
+            }
+
+            isBindingUpdatedOutsideOfControl = false;
+        }
+
+        private IReadOnlyList<ILocationListItem> BuildLocationListItems(IAnalysisIssueFlowVisualization flow)
```

Comment:
I think this method deserves a comment explaining how we're flattening and grouping a hierarchy into a flat list of items of different types, as it's not obvious why the code is doing what it's doing.